src/pytest_plugins/consume/consume.py

```diff
@@ -68,6 +70,51 @@ def download_and_extract(url: str, base_directory: Path) -> Path:
     return extract_to / "fixtures"
 
 
+@dataclass
+class SimLimitBehavior:
+    """Stores options derived from the `--sim.limit` argument."""
+
+    collectonly: bool = False
+    pattern: str = ".*"
+
+
+def process_sim_limit(pattern: str) -> SimLimitBehavior:
+    """
+    Process the `--sim.limit` argument to determine test filtering behavior.
+
+    The function handles three cases:
+    1. If the value is exactly "collectonly", enable collection mode without
+        filtering.
+    2. If the value starts with "collectonly:", enable collection mode and use the
+        rest as a regex pattern.
+    3. If the (possibly transformed) value starts with "id:", treat the rest as a
```

Comment:
I think one thing that is not completely clear is whether something like `collectonly:id:some_test` is valid.